src/humanloop/decorators/prompt.py

```
@@ -0,0 +1,112 @@
+            with get_tracer().start_as_current_span(str(uuid.uuid4())) as span:
+                trace_metadata = get_trace_context()
+
+                if trace_metadata:
+                    # We are in a Flow context
+                    write_to_opentelemetry_span(
+                        span=span,
+                        key=HL_TRACE_METADATA_KEY,
+                        value={**trace_metadata, "is_flow_log": False},
+                    )
+                    # Set current Prompt to act as parent for Logs nested underneath
+                    push_trace_context(
+                        {
+                            **trace_metadata,
+                            "trace_parent_id": span.get_span_context().span_id,
+                            "is_flow_log": False,
+                        },
+                    )
+
+                write_to_opentelemetry_span(
+                    span=span,
+                    key=HL_FILE_OT_KEY,
+                    value={
+                        "path": path if path else func.__name__,
+                        "prompt": {
+                            "template": template,
+                            "temperature": temperature,
+                            "top_p": top_p,
+                            "presence_penalty": presence_penalty,
+                            "frequency_penalty": frequency_penalty,
+                            "model": model,
+                            "endpoint": endpoint,
+                            "provider": provider,
+                            "max_tokens": max_tokens,
+                            "stop": stop,
+                        },
+                    },
+                )
+
+                try:
+                    output = func(*args, **kwargs)
+                except Exception as e:
+                    # TODO Some fails coming from here, they result in a fast end or duplicate
+                    # spans outputted to the Humanloop API
+                    print(e)
```

Remove print statement. Should we have a logger defined? 

Also this TODO is a blocker. Where are the tests for this in the SDK?

---

Should we catch the error in the first place? any intercept exception stems from developer written code, and I don't want it to fail silently

---

We should be logging errors if there are unhandled errors from code that is being wrapped

---

I am now logging the errors for the user and for the log output is set to None while error is the stringified exception